### PR TITLE
Closes #270 (Fixes HashMapExtensions + Tests)

### LINKED ISSUE #270: [0.0.4-hf1] Take in changes from #255
Bring in the changes from #255 - prepare for release as `0.0.4-hf1`

### PR DESCRIPTION
**Pull request by iambmelt**
See #255 / related #256 / origin #270

### CODE REVIEW

File: common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public void testURLFormDecodeNegative() throws IllegalArgumentException,
         final String methodName = "urlFormDecode";
         Object object = ReflectionUtils.getNonPublicInstance("com.microsoft.identity.common.adal.internal.util.HashMapExtensions");
```
Comment (shoatman) [inline common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: This method is public now I think.

Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: Great - thanks. I'll open up a work item and we can apply this change forwards

Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: Work item is here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/272

Thanks @shoatman

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java [hunk 2]
```diff
@@ -71,22 +74,38 @@ private HashMapExtensions() {
...
+                                "Encoding format is not supported",
+                                e
+                        );
+                        continue;
                     }
-
-                    if (!StringExtensions.isNullOrBlank(key)
-                            && !StringExtensions.isNullOrBlank(value)) {
-                        result.put(key, value);
+                } else if (elements.length == 1) {
```
Comment (titorodi) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java]: Woo

File: common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java [hunk 3]
```diff
@@ -61,7 +61,7 @@ public void testURLFormDecodeNegative() throws IllegalArgumentException,
...
         Method m = ReflectionUtils.getTestMethod(object, methodName, String.class);
-        HashMap<String, String> result = (HashMap<String, String>) m.invoke(object, "nokeyvalue");
+        HashMap<String, String> result = (HashMap<String, String>) m.invoke(object, "");
```
Comment (titorodi) [inline common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: Is this worth commenting that this checks for 'CertAuthorities='?

File: common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java [hunk 4]
```diff
@@ -124,5 +120,14 @@ public void testURLFormDecodePositive() throws IllegalArgumentException,
         assertTrue(result.containsKey("d"));
         assertTrue(result.containsValue("f"));
         assertTrue(result.size() == 1);
+
+        // This test verifies key/value query params can support empty values
```
Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: @titorodi ^^

File: common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java (hunk 3 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensionTests.java]: Resolved -- see below

**Pull request merged**